python/setup.py

```diff
@@ -385,17 +385,19 @@ def _bundle_arrow_cpp(self, build_prefix, build_lib):
                 build_prefix, build_lib,
                 "{}_regex".format(self.boost_namespace),
                 implib_required=False)
-        if sys.platform == 'win32':
-            # zlib uses zlib.dll for Windows
-            zlib_lib_name = 'zlib'
-            move_shared_libs(build_prefix, build_lib, zlib_lib_name,
-                             implib_required=False)
-            if self.with_flight:
-                # DLL dependencies for gRPC / Flight
-                for lib_name in ['libcrypto-1_1-x64',
-                                 'libssl-1_1-x64']:
-                    move_shared_libs(build_prefix, build_lib, lib_name,
-                                     implib_required=False)
+        # TODO(kszucs): consider to bundle the following shared libraries
```

Comment:
We can choose to bundle security sensitive shared libraries instead of linking them statically.

Reply:
In the case of OpenSSL, if we bundled a shared version the version in the main system will be preferred. Not sure though where this is updated from. In `conda-forge` we explicitly have a Python setup that takes care that one is not used.

Reply:
> In the case of OpenSSL, if we bundled a shared version the version in the main system will be preferred.

Shouldn't we prefer the system one over the bundled one?

> Not sure though where this is updated from.

You mean the bundled or the system one?

Reply:
Not sure about this. I don't know where the system one is coming from.

Reply:
How about keeping it as is, linking these libraries statically?

Reply:
We should keep as-is until someone visits with a strong *and*  reasonable opinion.

Reply:
Ok, I'm going to clean up the commented section used for bundling from `python/CMakeLists.txt` and `setup.py` then.

Reply:
And create a follow-up JIRA.

Reply:
https://issues.apache.org/jira/browse/ARROW-11186